services/user-management/src/main/java/com/foodie/user/utils/FoodieUserDetails.java

```diff
@@ -0,0 +1,63 @@
+package com.foodie.user.utils;
+
+import com.foodie.user.model.User;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import java.util.Collection;
+import java.util.UUID;
+
+public class FoodieUserDetails implements UserDetails {
+
+    private User user;
+
+    public FoodieUserDetails(User user) {
+        this.user = user;
+    }
+
+    public User getUser() {
```

Comment:
Use lombok?

Reply:
Did not use it cause of this custom detail

@Override
    public String getUsername() {
        return user.getEmail();
    }

Reply:
You can still have it in the class. Your custom details will override the ones Lombok created